docs: Add Configuration tutorial (#495)
unoplatform/uno.extensions: doc/Learn/Tutorials/Configuration/HowTo-Configuration.md

---
uid: Learn.Tutorials.Configuration.HowToConfiguration
---
# How-To: Get Started with Configuration

> Configuration in Uno.Extensions provides you with a consistent way to view configuration data from a number of distinct sources. This pattern is not intended to allow for programmatic writes by itself, but instead offers a single representation of any number of providers or sections.

Reviewer: We probably should be explaining what a "section" is.

> Configuration in Uno.Extensions provides you with a consistent way to view configuration data from a number of distinct sources. This pattern is not intended to allow for programmatic writes by itself, but instead offers a single representation of any number of providers or sections.

Reviewer 2: ```suggestion
Configuration in Uno.Extensions provide you with a consistent way to view configuration data from a number of distinct sources. This pattern is not intended to allow for programmatic writing by itself, but instead offers a single representation of any number of providers or sections.
```

## Step-by-steps

### 1. Specify configuration information to load on `IConfigBuilder`

* Uno.Extensions apps specify which configuration information to load by calling the `UseConfiguration()` extension method for `IHostBuilder`.

* Use the `EmbeddedSource<T>()` extension method on `IConfigBuilder` to load configuration information from an assembly type you specify:

    ```csharp
    public App()
    {
        Host = UnoHost
            .CreateDefaultBuilder()
            .UseConfiguration(configure: configBuilder =>
                configBuilder
                    // Load configuration information from appsettings.json
                    .EmbeddedSource<App>()
            )
    ...
    ```

> * By default, this method will extract values from an embedded resource called `appsettings.json`, unless you optionally denote a different file name. The string you pass into the extension method will be concatenated in between `appsettings` and its file extension. For instance, the following will also retrieve values from the file `appsettings.platform.json` embedded inside the `App` assembly:

Reviewer: ```suggestion
* By default, this method will extract values from an embedded resource (using the `EmbeddedResource` file build action) called `appsettings.json`, unless you optionally denote a different file name. The string you pass into the extension method will be concatenated in between `appsettings` and its file extension. For instance, the following will also retrieve values from the file `appsettings.platform.json` embedded inside the `App` assembly:
```

> * By default, this method will extract values from an embedded resource called `appsettings.json`, unless you optionally denote a different file name. The string you pass into the extension method will be concatenated in between `appsettings` and its file extension. For instance, the following will also retrieve values from the file `appsettings.platform.json` embedded inside the `App` assembly:

Reviewer 2: ```suggestion
* By default, this method will extract values from an embedded resource called `appsettings.json`, unless you optionally denote a different file name. The string you pass into the extension method will be concatenated in-between `appsettings` and its file extension. For instance, the following will also retrieve values from the file `appsettings.platform.json` embedded inside the `App` assembly:
```
